### ISSUE TITLE
Setting "price" or "price override" field raises an error

### ISSUE THREAD

**Issue by maarcingebala** [mirumee/saleor#4134]
### Steps to reproduce the problem
1. Set a product price or variant's "price override" field to zero (either in Dashboard 1.0 or 2.0).
2. Go to product details page (both dashboards).
3. An error is raised:

```
"Traceback (most recent call last):",
            "  File \"/Users/marcin/.pyenv/versions/saleor3.7/lib/python3.7/site-packages/promise/promise.py\", line 487, in _resolve_from_executor",
            "    executor(resolve, reject)",
            "  File \"/Users/marcin/.pyenv/versions/saleor3.7/lib/python3.7/site-packages/promise/promise.py\", line 754, in executor",
            "    return resolve(f(*args, **kwargs))",
            "  File \"/Users/marcin/.pyenv/versions/saleor3.7/lib/python3.7/site-packages/graphql/execution/middleware.py\", line 76, in make_it_promise",
            "    return next(*args, **kwargs)",
            "  File \"/Users/marcin/.pyenv/versions/saleor3.7/lib/python3.7/site-packages/graphql_jwt/decorators.py\", line 31, in wrapper",
            "    return func(info.context, *args, **kwargs)",
            "  File \"/Users/marcin/.pyenv/versions/saleor3.7/lib/python3.7/site-packages/graphql_jwt/decorators.py\", line 42, in wrapper",
            "    return f(*args, **kwargs)",
            "  File \"/Users/marcin/mirumee/saleor/saleor/graphql/product/types/products.py\", line 480, in resolve_margin",
            "    _, margin = get_product_costs_data(self)",
            "  File \"/Users/marcin/mirumee/saleor/saleor/product/utils/costs.py\", line 14, in get_product_costs_data",
            "    costs_data = get_cost_data_from_variants(variants)",
            "  File \"/Users/marcin/mirumee/saleor/saleor/product/utils/costs.py\", line 34, in get_cost_data_from_variants",
            "    costs_data = get_variant_costs_data(variant)",
            "  File \"/Users/marcin/mirumee/saleor/saleor/product/utils/costs.py\", line 44, in get_variant_costs_data",
            "    margin = get_margin_for_variant(variant)",
            "  File \"/Users/marcin/mirumee/saleor/saleor/product/utils/costs.py\", line 61, in get_margin_for_variant",
            "    percent = round((margin / base_price) * 100, 0)",
            "  File \"/Users/marcin/.pyenv/versions/saleor3.7/lib/python3.7/site-packages/prices/money.py\", line 74, in __truediv__",
            "    return self.amount / other.amount",
            "decimal.DivisionByZero: [<class 'decimal.DivisionByZero'>]"
```

I'm not sure if it makes sense to set a zero price, but both dashboards allow that. I guess the `get_margin_for_variant` function which handles margin calculation should be anyway fixed to handle the case where the price is equal to zero.

**Comment by MahmoudRizk** [comment]
I want to work on this issue @maarcingebala  @NyanKiyoshi

**Comment by NyanKiyoshi** [comment]
Hi! Sure, nobody is assigned to it yet. Feel free to open a PR!

**Comment by NyanKiyoshi** [comment]
Closed by #4134.
